File: 30 Days of Code/Day 13 Abstract Classes/MyBook.java
import java.util.*;

abstract class Book {
	String title;
	String author;

	Book(String title, String author) {
		this.title = title;
		this.author = author;
	}

	abstract void display();
}

class MyBook extends Book {
	Integer price;

	MyBook(String title, String author, Integer price) {
		super(title, author);
		this.price = price;
	}

	void display() {
		System.out.println("Title: " + this.title);
		System.out.println("Author: " + this.author);
		System.out.println("Price: " + this.price);
	}

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		String title = input.nextLine();
		String author = input.nextLine();
		int price = input.nextInt();

		Book book = new MyBook(title, author, price);
		book.display();
	}
}

File: 30 Days of Code/Day 13 Abstract Classes/my_book.cpp
#include <bits/stdc++.h>
using namespace std;

class Book
{
	public:
		Book(string title, string author):
			title_(title),
			author_(author)
		{}

		virtual void display()=0;

	protected:
		string title_;
		string author_;
};

class MyBook: public Book
{
	public:
		MyBook(string title, string author, int price):
			Book(title, author),
			price_(price)
		{}

		virtual void display() {
			cout << "Title: " + title_ << endl;
			cout << "Author: " + author_ << endl;
			cout << "Price: " << price_;
		}

	private:
		int price_;
};

int main()
{
	string title = "", author = "";
	int price = 0;
	getline(cin, title);
	getline(cin, author);
	cin >> price;

	MyBook *novel = new MyBook(title, author, price);
	novel->display();
	delete novel;
}